src/main/host/descriptor/socket/inet/udp.rs

```diff
@@ -887,17 +933,43 @@ impl UdpSocket {
     }
 }
 
-/// A buffer of packets.
-struct PacketBuffer {
-    /// The packets in the buffer.
-    buffer: std::collections::LinkedList<PacketRc>,
+/// Non-payload data for a message in the send buffer.
+#[derive(Debug)]
+struct MessageSendHeader {
+    /// The source address (typically the bind address). The application can theoretically use
+    /// `IP_PKTINFO` to set a per-message source address.
+    src: SocketAddrV4,
+    /// The destination address (for example the peer).
+    dst: SocketAddrV4,
+    /// The priority for the packet that we'll create in the future, given to us by the host.
+    packet_priority: u64,
```

Comment:
Optional - maybe it's be worth creating a type for packet priorities? Maybe it's not worth the fuss while much of the code is still in C though.

Reply:
Added a `FifoPacketPriority` typedef.